internal/epoch/epoch_utils.go

```diff
@@ -91,3 +93,57 @@ func deletionWatermarkFromBlobID(blobID blob.ID) (time.Time, bool) {
 
 	return time.Unix(unixSeconds, 0), true
 }
+
+type intRange struct {
+	lo, hi int
+}
+
+func (r intRange) length() uint {
+	return uint(r.hi - r.lo)
+}
+
+func (r intRange) isEmpty() bool {
+	return r.length() == 0
+}
+
+var errNonContiguousRange = errors.New("non-contiguous range")
+
+// constants from the standard math package.
+const (
+	//nolint:gomnd
+	intSize = 32 << (^uint(0) >> 63) // 32 or 64
+
+	maxInt = 1<<(intSize-1) - 1
+	minInt = -1 << (intSize - 1)
```

Comment:
Any reason to create consts here instead of using [math.MinMaxConsts](https://pkg.go.dev/math#pkg-constants)?

Reply:
To avoid taking a dependency here on the math package just for importing 2 constants.
A test was included for sanity. The test does take a dependency on the math package.

Reply:
In kopia we already have `import "math"` in multiple places  and `epoch` package is under `internal` and couldn't be used as a standalone package outside the project. So we can save 10+ lines of the code with one import. :)